File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> umap;
        int i = 0, diff = 0, n = nums.size();
        for (i = 0; i<n; i++){
            diff = target-nums[i];
            if (umap.find(diff)!=umap.end())
                break;
            else
                umap[nums[i]] = i;
        }
        return {i,umap[diff]};
    }
};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dict_ = dict();
        ans = [0,0];
        
        for i in range(len(nums)):
            goal = target-nums[i]
            if (goal in dict_):
                ans[0] = dict_[goal]
                ans[1] = i
                return ans
            dict_[nums[i]] = i
            
        return ans
        
        
        